test/discrete/test_bdq.py

```diff
@@ -115,7 +115,8 @@ def test_bdq(args: argparse.Namespace = get_args()) -> None:
     )
     test_collector = Collector(policy, test_envs, exploration_noise=False)
     # policy.set_eps(1)
-    train_collector.collect(n_step=args.batch_size * args.training_num, reset_before_collect=True)
+    train_collector.reset()
+    train_collector.collect(n_step=args.batch_size * args.training_num)
```

Comment:
This shouldn't have been necessary as `reset_before_collect=True`. No need to rollback, but just FYI. The tests did run through in CI, so the problem was only in some examples (where stuff was collected before training started)

Reply:
I was also unsure about this one. When is the right time to pass the `reset_before_collect` and when should one be calling reset directly? I assumed that after instantiation I want to properly reset once.

Reply:
There's really no difference, it was just a convenience flag. We could also remove it if it causes confusion